### DISCUSSION TITLE
Method not allowed

### DISCUSSION THREAD

**Discussion by ocersnguyend** [Quansight/ragna#156] [General]
It took a long time to process one file, over night.  But now when I try out a query I'm getting 
Client error '400 Bad Request' for url 'http://127.0.0.1:31476/chats/0b1c66f9-0542-461e-b239-c85db93021e5/answer?prompt=hello'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
The response from the endpoint is 
{"detail":"Method Not Allowed"}
Please advise.
Thank you!

**Comment by pmeier** [comment]
How did you run the UI? `ragna ui` or did you use `ragna ui --config ...`? If it's the latter, could you post the  config you used?

**Reply by ocersnguyend** [reply]
Just ragna ui

**Reply by pmeier** [reply]
Is the API still running? What do you see when running `curl localhost:31476`? You can also visit this in a browser.

**Reply by ocersnguyend** [reply]
"0.1.0"

**Reply by pmeier** [reply]
Just from this, it is hard to debug.

- What source storage and assistant did you select?
- When running `ragna ui` without a config, you should have a database at `~/.cache/ragna/ragna.db`. Is that the case?
- Have you tried reloading the UI?

**Reply by ocersnguyend** [reply]
Yes, I know it must be hard to debug like this.  I'm trying on my end to see if I can get any useful info for you.  But I'm not seeing anything.
This is the ragna.toml it generated.  Note, I've tried the default assistant as well.
local_cache_root = "C:\\Users\\nguyend\\.cache\\ragna"

[core]
queue_url = "C:\\Users\\*******\\.cache\\ragna\\queue"
document = "ragna.core.LocalDocument"
source_storages = ["ragna.source_storages.Chroma", "ragna.source_storages.RagnaDemoSourceStorage", "ragna.source_storages.LanceDB"]
assistants = ["ragna.assistants.Gpt35Turbo16k"]

[api]
url = "http://127.0.0.1:31476"
database_url = "sqlite:///C:\\Users\\******\\.cache\\ragna/ragna.db"
authentication = "ragna.core.RagnaDemoAuthentication"
upload_token_secret = "*********************************"
upload_token_ttl = 300

[ui]
url = "http://127.0.0.1:31477"

when I start a new chat after dropping a file for upload, it just spins and spins and spins.  No status indication or anything.  I left it overnight and it was still spinning.  I had to ctrl-c to stop the server.  When I fire the UI back up, the chat history was there.
Could be because I had to break the execution that something is not completely closed out.
Since, I've tried different assistants and different files.  All of them just spin and I have to break after a long period of time.  Is there a switch to show what it's doing?
Thanks for helping by the way!

**Reply by pmeier** [reply]
> when I start a new chat after dropping a file for upload, it just spins and spins and spins. No status indication or anything. 

That seems like something gone wrong and the UI doesn't react to it. One thing that we improved since the initial release is properly forwarding the log messages from the underlying processes to the main process that runs the UI (#123). The patch is included in the latest pre-release. Could you do me a favor and do `pip install --upgrade --pre 'ragna[all]'` and try again? Note that the config format has changed slightly. From what I see above, it should only be deleting the `upload_token_*` entries. If there is something else it complains about, delete the file and run `ragna init` to create a fresh one.

If you still see the same behavior in the UI, you should now see an error in the console that it was started from. If that is the case, please post it here.

**Reply by ocersnguyend** [reply]
There's a authentication dialog now.  What should I use?
If I click on sign in without then I get a bunch of error starting with
Task exception was never retrieved
future: <Task finished name='Task-145' coro=<WebSocketProtocol13.write_message.<locals>.wrapper() done, defined at C:\Users\*****\AppData\Roaming\Python\Python311\site-packages\tornado\websocket.py:1085> exception=WebSocketClosedError()>
Traceback (most recent call last):
  File "C:\Users\*****\AppData\Roaming\Python\Python311\site-packages\tornado\websocket.py", line 1087, in wrapper
    await fut
tornado.iostream.StreamClosedError: Stream is closed

But it does get me to the New Chat screen.  When I drop a file and tried to start a conversation, I get
requests.exceptions.SSLError: HTTPSConnectionPool(host='chroma-onnx-models.s3.amazonaws.com', port=443): Max retries exceeded with url: /all-MiniLM-L6-v2/onnx.tar.gz (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)')))

I think this could be on our side as our network is very locked down.  At least now I know what it's trying to reach.  I will see if I can get the URL whitelist and see if I can continue on.
Thanks!  The error output really helps!

**Reply by pmeier** [reply]
> There's a authentication dialog now. What should I use?

It implements the demo auth documented here: https://ragna.chat/en/stable/references/python-api/#ragna.core.RagnaDemoAuthentication. By default it requires the username and password to match.

> If I click on sign in without then I get a bunch of error starting with

Welp, I guess not putting anything as username and password fulfills the requirement of the two matching up :upside_down_face: Let me add an issue for that.

> I think this could be on our side as our network is very locked down. At least now I know what it's trying to reach. I will see if I can get the URL whitelist and see if I can continue on.

That is on your side. We are trying to load a third-party embedding model. More specifically https://github.com/chroma-core/onnx-embedding. I guess you could also compile it yourself, but that sounds more complicated :shrug:
